scio-core/src/main/scala/com/spotify/scio/coders/instances/GuavaCoders.scala

```diff
@@ -22,7 +22,7 @@ import com.spotify.scio.coders.Coder
 import com.google.common.{hash => g}
 import org.apache.beam.sdk.coders.CustomCoder
 
-class GuavaBloomFilterCoder[T](implicit val funnel: g.Funnel[T])
+class GuavaBloomFilterCoder[T](implicit val funnel: g.Funnel[_ >: T])
```

Comment:
`BLoomFilter[String]` can be created from a `Funnel[CharSequence]` (expected funnel is `<? extends T>`).
Associated coder should also be able to be constructed with such funnel